pydantic/types.py

```diff
@@ -1113,32 +1116,62 @@ def to(self, unit: str) -> float:
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ DATE TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 if TYPE_CHECKING:
+    from datetime import date
```

Comment:
This is not correct, we should just do

```py
from datetime import date
```

above in the module scope, then use `date` etc. everywhere. This will reduce the size of the change significantly.